sycl/include/sycl/ext/intel/experimental/esimd/simd.hpp

```diff
@@ -60,13 +60,19 @@ class simd
     __esimd_dbg_print(simd(const SimdT &RHS));
   }
 
-  // Broadcast constructor with conversion.
+  /// Broadcast constructor with conversion.
   template <typename T1,
             class = std::enable_if_t<detail::is_vectorizable_v<T1>>>
   simd(T1 Val) : base_type((Ty)Val) {
     __esimd_dbg_print(simd(T1 Val));
   }
 
+  /// Construct from simd_view.
+  template <int ViewedSimdLength, typename RegionT,
```

Comment:
Thanks!
Seems like this is almost complete copy-paste of the simd_obj_impl variant. I missed that in the first review, unfortunately. Can duplication be somehow avoided? E.g. this simd variant just refers to the simd_obj_impl variant?

One more thing that I missed is simd_mask - it should also have this kind of constructor. I wonder if it is enough just to define that constructor in a generic way so that it is automatically inherited from simd_obj_impl by both simd and simd_mask w/o the need to define this constructor explicitly in them?

Reply:
Technically it can be deleted, I'll do that.

Need to add though that the deduction  guide needs to be defined for 'simd' class, not for 'simd_obj_impl'.
Having something like this will now work for the assignment: "simd_view SV = V; simd V1 = SV":

> // Deduction guide for 'simd_obj_impl' constructor from a 'simd_view' object.
> // It helps to infer the template parameter 'N' of the class 'simd_obj_impl'.
> template <typename T, int ViewedSimdLength, typename RegionT>
> simd_obj_impl(simd_view<simd<T, ViewedSimdLength>, RegionT>) -> simd_obj_impl<T, RegionT::length, simd<T, RegionT::length>>;
>

Reply:
One more note: the mentioned 'simd' constructor can be deleted only because of this statement in simd class: 

> using base_type::base_type;

The same for 'simd -> simd_view' is not doable because 'simd_view' does not have such 'using' statement and the following ctor cannot be removed:
https://github.com/intel/llvm/blob/sycl/sycl/include/sycl/ext/intel/experimental/esimd/simd_view.hpp#L76

Reply:
Regarding the 'simd_mask' and doing the arguments deduction for it...
Currently, 'simd_mask' is declared as an alias template, not as a template class and thus the template args deduction guides do not work for it.

```
simd_mask<16> m16 = 0;
auto mask_view8 = m16.select<8, 1>(0);

detail::simd_mask_impl m8 = mask_view8; // Works well with some changes in headers
simd_mask m8 = mask_view8; // Does not work. The error is: "error: alias template 'simd_mask' requires template arguments; argument deduction only allowed for class templates"

```

I don't know why simd_mask is not defined as a class instead of an alias to a class in detail namespace.
Anyway, tackling this problem for simd_mask deserves a separate PR, IMO.